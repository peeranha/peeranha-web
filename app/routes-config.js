/* eslint camelcase: 0 */
export const home = () => `/`;
export const faq = () => `/faq`;

export const profileView = id => `/users/${id}`;

export const profileViewActivityQuestions = id =>
  `/users/${id}#activity#questions`;

export const profileViewActivityAnswers = id => `/users/${id}#activity#answers`;

export const profileEdit = id => `/users/edit/${id}`;

export const userQuestions = id => `/users/${id}#questions`;
export const userAnswers = id => `/users/${id}#answers`;
export const userSettings = id => `/users/${id}/settings`;

export const questions = () => `/questions`;

export const uniqueAnswerId = answerId => `ans${answerId}`;

export const questionView = /* istanbul ignore next */ (id, answerId) =>
  answerId
    ? `/questions/${id}#${uniqueAnswerId(answerId)}`
    : `/questions/${id}`;

export const questionEdit = questionid => `/questions/${questionid}/edit`;

export const answerEdit = (questionid, answerid) =>
  `/questions/${questionid}/answers/${answerid}/edit`;

export const questionAsk = () => `/questions/ask`;

export const noAccess = () => `/no-access`;
export const feed = () => `/feed`;
export const communities = () => `/communities`;
export const tags = () => `/tags`;
export const users = () => `/users`;
export const appFaq = () => `/app/faq`;
export const support = () => `/support`;

export const communitiesCreate = () => `/communities/create`;
export const suggestedCommunities = () => `/communities/suggested`;
export const communityTags = communityid => `/communities/${communityid}/tags`;

export const suggestedTags = communityid =>
  `/communities/${communityid}/tags/suggested`;

export const tagsCreate = communityid =>
  `/communities/${communityid || 0}/tags/create`;

export const registrationStage = 'signup';

export const signup = {
  email: {
    step: 1,
    name: `/${registrationStage}/email`,
    scatter: false,
  },
  displayName: {
    step: 1,
    name: `/${registrationStage}/with-scatter/display-name`,
    scatter: true,
  },
  emailVerification: {
    step: 2,
    name: `/${registrationStage}/email-verification`,
    scatter: false,
  },
  haveEosAccount: {
    step: 3,
    name: `/${registrationStage}/i-have-eos-account`,
    scatter: false,
  },
  dontHaveEosAccount: {
    step: 3,
    name: `/${registrationStage}/i-dont-have-eos-account`,
    scatter: false,
  },
  almostDone: {
    step: 4,
    name: `/${registrationStage}/almost-done`,
    scatter: false,
  },
};

export const privacyPolicy = () => `/privacy-policy`;
