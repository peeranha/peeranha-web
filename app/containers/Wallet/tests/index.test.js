import { noAccess } from 'routes-config';
import { Wallet } from '../index';

const cmp = new Wallet();

beforeEach(() => {
  cmp.props = {
    balance: null,
    locale: 'en',
    account: null,
    match: { params: { id: 'id' } },
    history: { push: jest.fn() },
    getWeekStatDispatch: jest.fn(),
    weekStat: [],
    getWeekStatProcessing: false,
  };
});

describe('Wallet', () => {
  it('render', () => {
    expect(cmp.render()).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    it('@account is truthy', () => {
      cmp.props.account = 'account';

      expect(cmp.props.getWeekStatDispatch).toHaveBeenCalledTimes(0);
      expect(cmp.props.history.push).toHaveBeenCalledTimes(0);

      cmp.componentDidMount();

      expect(cmp.props.getWeekStatDispatch).toHaveBeenCalledTimes(1);
      expect(cmp.props.history.push).toHaveBeenCalledTimes(0);
    });

    it('@account is null', () => {
      cmp.props.account = null;

      expect(cmp.props.getWeekStatDispatch).toHaveBeenCalledTimes(0);
      expect(cmp.props.history.push).toHaveBeenCalledTimes(0);

      cmp.componentDidMount();

      expect(cmp.props.getWeekStatDispatch).toHaveBeenCalledTimes(1);
      expect(cmp.props.history.push).toHaveBeenCalledTimes(1);
      expect(cmp.props.history.push).toHaveBeenCalledWith(noAccess());
    });
  });
});
