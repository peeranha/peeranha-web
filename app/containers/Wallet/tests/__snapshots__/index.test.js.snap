// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet render 1`] = `
<div>
  <UNDEFINED
    description="Wallet | Description"
    index={false}
    language="en"
    title="Wallet"
  />
  <UNDEFINED
    account={null}
    balance={null}
    getWeekStatProcessing={false}
    locale="en"
    userId="id"
    weekStat={Array []}
  />
</div>
`;
